WordPress/gutenberg
# Make zip-based URL parsing more general

## Description
Zip URLs aren't parsed quite right in wp-env. For example, given a URL "https://github.com/CalderaWP/Caldera-Forms/files/5002105/ninja-forms.zip", the basename of the plugin will be "github.com%2FCalderaWP%2FCaldera-Forms%2Ffiles%2F5002105%2Fninja-forms". These characters are not supported as pathnames in JS and CSS, so the plugin would be broken.

The fix is to make our zip-parsing behavior more general. Now, given the same input, the zip result will be `ninja-forms`. This behavior was already used for sources from wordpress.org, but I'm just extending it to support other providers too.

While troubleshooting this issue, I also uncovered another where another type of zip wasn't supported. Basically, some zips will be created such that the zip _is_ the container of the plugin. Others, like wordpress.org, will create a zip such that the container of the plugin is a subdirectory of the zip. I added logic to make sure we handle both cases. Previously, we only handled the latter case.

cc @Shelob9 

Resolves #24291 

## How has this been tested?
1. Checkout this branch in gutenberg
2. In your gutenberg wp-env override, add a URL to a .zip plugin which is not on wordpress.org. I have uploaded this one you can use to test: `https://srv-store1.gofile.io/download/F1zUEb/editing-toolkit-build-archive.zip`
3. Run `npx wp-env start`. 
4. The plugin should be downloaded without the garbage characters in the path.

Additionally, unit tests have been created to verify the behavior.

## Screenshots 

## Types of changes
Bug fix

## Checklist:
